src/Host.cpp

```diff
@@ -384,10 +392,10 @@ std::tuple<bool, QString, QString, QString, QString> Host::saveProfile()
         XMLexport writer(this);
         writer.exportHost(&file_xml);
         file_xml.close();
-        saveModules(1);
-        return std::make_tuple(true, QString(), QString(), QString(), QString());
+        saveModules(syncModules ? 1 : 0);
+        return std::make_tuple(true, filename_xml, QString(), QString());
     } else {
-        return std::make_tuple(false, QLatin1String("Couldn't save %1 to %2 because: %3"), getName(), filename_xml, file_xml.errorString());
+        return std::make_tuple(false, getName(), filename_xml, file_xml.errorString());
```

Comment:
I still think that second argument is bogus, the caller is going to know what the profile is called (it too, can use `Host::getName()`) so why is it being put into that tuple at all?